### PR TITLE
Add script that suggest upgrades from pypi

### PR DESCRIPTION
**Pull request by oyvindeide**

### CODE REVIEW

File: komodo/check_up_to_date_pypi.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+def get_release(yaml, fname):
+    with open(fname, "r") as fin:
+        result = yaml.load(fin)
+    return result
+
+
+def get_packages_pypi(package_names):
+    return [
+        (package, requests.get(f"https://pypi.python.org/pypi/{package}/json"))
```
Comment (sondreso) [inline komodo/check_up_to_date_pypi.py:29]: Just out of curiosity, how long does this take? :sweat_smile:

File: komodo/check_up_to_date_pypi.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+
+def yaml_parser():
+    yaml = ruamel.yaml.YAML()
+    yaml.indent(mapping=2, sequence=4, offset=2)
+    yaml.preserve_quotes = True
+    return yaml
+
+
+def get_release(yaml, fname):
```
Comment (sondreso) [inline komodo/check_up_to_date_pypi.py:21]: maybe rename to load_from_file (or something similar)?

File: komodo/check_up_to_date_pypi.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    return pypi_packages
+
+
+def get_upgrade_proposals(release_file, repository_file, python_version, propose_upgrade):
+    yaml = yaml_parser()
+    releases = get_release(yaml, release_file)
+    repository = get_release(yaml, repository_file)
+
+    pypi_packages = get_pypi_packages(releases, repository)
+    pypi_response = get_packages_pypi(pypi_packages)
```
Comment (sondreso) [inline komodo/check_up_to_date_pypi.py:83]: The naming here is slightly confusing, we could call the bottom one get_pypi_info or something like that instead?

File: komodo/check_up_to_date_pypi.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            )
+        if pypi_latest_version != komodo_version:
+            upgrade_proposals[name] = {"previous": str(komodo_version), "suggested": str(pypi_latest_version)}
+
+    if propose_upgrade and upgrade_proposals:
+        for name, version in upgrade_proposals.items():
+            suggested_version = version["suggested"]
+            if suggested_version not in repository[name]:
+                repository[name].update(
+                    {suggested_version: copy.deepcopy(repository[name][releases[name]])}
```
Comment (sondreso) [inline komodo/check_up_to_date_pypi.py:105]: If I have understood this correct, then maybe this reads a little bit better? :thinking: 
```suggestion
                    {suggested_version: copy.deepcopy(repository[name][version["previous"]])}
```

**Review by sondreso — COMMENTED** [review]
Very nice :clap:  Other then my small comments/question this looks really good!

File: komodo/check_up_to_date_pypi.py (hunk 1 above)
Comment (oyvindeide) [reply komodo/check_up_to_date_pypi.py:29]: Shorter than you (or I at least) would think. I started shooting of more at the time, but it was not really needed I felt.

File: komodo/check_up_to_date_pypi.py (hunk 4 above)
Comment (oyvindeide) [reply komodo/check_up_to_date_pypi.py:105]: I agree that would be better, but the version there has been sanitized (to remove `+py3` or what we might find appropriate to add), so will not fit directly. Will try to improve readability.

Comment (oyvindeide) [reply komodo/check_up_to_date_pypi.py:105]: And at the same time, add a test for that `+py3` thing

**Comment by oyvindeide** [comment]
Did a small refactor to improve readability, and added a full integration test.

**Review by sondreso — APPROVED** [review]
LGTM!

File: komodo/check_up_to_date_pypi.py (hunk 1 above)
Comment (oyvindeide) [reply komodo/check_up_to_date_pypi.py:33]: Dont get me wrong, it is quite slow, but acceptably slow (in my opinion)

**Pull request merged**